go/parquet/metadata/column_chunk.go

```diff
@@ -0,0 +1,385 @@
+// for constructing or duplicating column metadata
+func (c *ColumnChunkMetaDataBuilder) Contents() *format.ColumnChunk { return c.chunk }
+
+func (c *ColumnChunkMetaDataBuilder) init(chunk *format.ColumnChunk) {
+	c.chunk = chunk
+	if !c.chunk.IsSetMetaData() {
+		c.chunk.MetaData = format.NewColumnMetaData()
+	}
+	c.chunk.MetaData.Type = format.Type(c.column.PhysicalType())
+	c.chunk.MetaData.PathInSchema = schema.ColumnPathFromNode(c.column.SchemaNode())
+	c.chunk.MetaData.Codec = format.CompressionCodec(c.props.CompressionFor(c.column.Path()))
+}
+
+func (c *ColumnChunkMetaDataBuilder) SetFilePath(val string) {
+	c.chunk.FilePath = &val
+}
+
+// Descr returns the associated column descriptor for this column chunk
+func (c *ColumnChunkMetaDataBuilder) Descr() *schema.Column { return c.column }
+
+func (c *ColumnChunkMetaDataBuilder) TotalCompressedSize() int64 {
+	// if this column is encrypted, after Finish is called, the MetaData
+	// field is set to nil and we store the compressed size so return that
+	if c.chunk.MetaData == nil {
+		return c.compressedSize
+	}
+	return c.chunk.MetaData.GetTotalCompressedSize()
+}
+
+func (c *ColumnChunkMetaDataBuilder) SetStats(val EncodedStatistics) {
+	c.chunk.MetaData.Statistics = val.ToThrift()
+}
+
+// Finish finalizes the metadata with the given offsets,
+// flushes any compression that needs to be done, and performs
+// any encryption if an encryptor is provided.
+func (c *ColumnChunkMetaDataBuilder) Finish(nvalues, dictPageOffset, indexPageOffset, dataPageOffset, compressed, uncompressed int64, hasDict, dictFallback bool, dictEncodingStats, dataEncodingStats map[parquet.Encoding]int32, metaEncryptor encryption.Encryptor) error {
+	if dictPageOffset > 0 {
+		c.chunk.MetaData.DictionaryPageOffset = &dictPageOffset
+		c.chunk.FileOffset = dictPageOffset + compressed
+	} else {
+		c.chunk.FileOffset = dataPageOffset + compressed
+	}
+
+	c.chunk.MetaData.NumValues = nvalues
+	if indexPageOffset >= 0 {
+		c.chunk.MetaData.IndexPageOffset = &indexPageOffset
+	}
+
+	c.chunk.MetaData.DataPageOffset = dataPageOffset
+	c.chunk.MetaData.TotalUncompressedSize = uncompressed
+	c.chunk.MetaData.TotalCompressedSize = compressed
+
+	thriftEncodings := make([]format.Encoding, 0, 3)
+	if hasDict {
+		thriftEncodings = append(thriftEncodings, format.Encoding(c.props.DictionaryIndexEncoding()))
+		if c.props.Version() == parquet.V1 {
+			thriftEncodings = append(thriftEncodings, format.Encoding_PLAIN)
+		} else {
+			thriftEncodings = append(thriftEncodings, format.Encoding(c.props.DictionaryPageEncoding()))
+		}
+	} else { // no dictionary
+		thriftEncodings = append(thriftEncodings, format.Encoding(c.props.EncodingFor(c.column.Path())))
+	}
+
+	thriftEncodings = append(thriftEncodings, format.Encoding(parquet.Encodings.RLE))
+	// Only PLAIN encoding is supported for fallback in V1
+	// TODO(zeroshade): Use user specified encoding for V2
```

Comment:
note: https://github.com/apache/arrow/pull/11031 recently deprecated "V2" as a standalone concept and now reference minor versions.

Reply:
So looking at that PR, it looks like the actual Parquet Format still only has the two values for version being 1 and 2 (with 2 now referring to "most recent version 2") and there's no way to actually *write* the minor version to the parquet file? Do I have that right? So the "fine grained" version handling is only for specific feature selection of writing, but they all still just write "2" to the parquet file metadata, right?

Reply:
Yes, that is my understanding.